lib/SILOptimizer/ARC/GlobalARCSequenceDataflow.cpp

```diff
@@ -254,8 +222,17 @@ bool ARCSequenceDataflowEvaluator::processBBBottomUp(
       RCIA, EAFI, BBState, FreezeOwnedArgEpilogueReleases, IncToDecStateMap,
       SetFactory);
 
-  // For each terminator instruction I in BB visited in reverse...
-  for (auto II = std::next(BB.rbegin()), IE = BB.rend(); II != IE;) {
+  auto II = BB.rbegin();
+  if (isa<TermInst>(*II)) {
+    auto *terminator = &cast<TermInst>(*II);
+    if (terminator == BB.getTerminator() &&
+        !isARCSignificantTerminator(terminator)) {
```

Comment:
It seems like there's some extra logic here. Why not just
```
if (!isARCSignificantTerminator(cast<TermInt>(*II))
 ++II;
```

Reply:
Done.